gravitee-am-service/src/main/java/io/gravitee/am/service/impl/DomainServiceImpl.java

```diff
@@ -489,6 +492,10 @@ public Completable delete(String domainId, User principal) {
                             )
                             // delete rate limit
                             .andThen(rateLimiterService.deleteByDomain(domain, DOMAIN))
+                            .andThen(passwordHistoryService.findByReference(ReferenceType.DOMAIN, domainId)
```

Comment:
For a single domain with thousands of users (or even more), we may have a lot a single delete request to do in order to clean the passwordHistory table. A single `deleteByDomain` method should be more efficient.